pkg/frontend/querymiddleware/request_validation_test.go

```diff
@@ -209,9 +209,13 @@ func TestCardinalityQueryRequestValidationRoundTripper(t *testing.T) {
 		},
 		{
 			// non-utf8 label name will be rejected even when we transition to UTF-8 label names
-			url:             cardinalityLabelValuesPathSuffix + "?label_names[]=\\xbd\\xb2\\x3d\\xbc\\x20\\xe2\\x8c\\x98",
+			url:             cardinalityLabelValuesPathSuffix + "?label_names[]=\xbd\xb2\x3d\xbc\x20\xe2\x8c\x98",
```

Comment:
why was the previous url still rejected even though it was ascii-only? was it because of the slashes? let's keep it as a test case in addition to the real UTF8 case you added/fixed 🤷

Reply:
i don't see any test cases for invalid strings

Reply:
On second look, you are right @dimitarvdimitrov. Could you add a case @juliusmh?

Reply:
I don't really understand this comment. We have cases for
Invalid names: `\xbd\xb2\x3d` , valid legacy/utf8 name: `foo` and valid utf8 name: `some.label`

Reply:
Ah yes, I wonder if I was looking at an outdated revision.